WordPress/Classes/Services/NotificationFilteringService.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+/// The service is created to support disabling WordPress notifications when Jetpack app is installed
+/// The service uses App Groups which allows Jetpack app to change the state of notifications flag and be later accessed by WordPress app
+/// This is a temporary solution to avoid duplicate notifications during the migration process from WordPress to Jetpack app
+/// This service and its usage can be deleted once the migration is done
+final class NotificationFilteringService {
+    private var notificationSettingsLoader: NotificationSettingsLoader
+    private var notificationsEnabled: Bool = false
+    private let allowDisablingWPNotifications: Bool
+    private let isWordPress: Bool
+    private let userDefaults = UserDefaults(suiteName: WPAppGroupName)
+
+    var wordPressNotificationsEnabled: Bool {
+        get {
+            guard let userDefaults = userDefaults,
+                  userDefaults.value(forKey: WPNotificationsEnabledKey) != nil else {
+                /// Treat this flag as enabled if it wasn't explicitly disabled
+                return true
+            }
+
+            return userDefaults.bool(forKey: WPNotificationsEnabledKey)
+        }
+
+        set {
+            userDefaults?.set(newValue, forKey: WPNotificationsEnabledKey)
+        }
+    }
+
+    init(notificationSettingsLoader: NotificationSettingsLoader = UNUserNotificationCenter.current(),
+         allowDisablingWPNotifications: Bool = FeatureFlag.allowDisablingWPNotifications.enabled,
```

Comment:
Hey @staskus 👋 
I noticed you've added a remote feature flag. Just wanted to note that directly using `FeatureFlag.enabled` does not return the remote value, instead, it returns the flag's default value.

We're aware that it's a bit confusing and we'll try to make it more intuitive in the next iteration.

For now, in order to retrieve the remote value, you'll have to do this:
```suggestion
         allowDisablingWPNotifications: Bool = RemoteFeatureFlagStore().value(for: FeatureFlag.allowDisablingWPNotifications),
```

Reply:
Thank you for letting me know! I'll make a change. I thought setting a `remoteKey` is enough.

Reply:
@hassaanelgarem 

To clarify, remote feature flags cannot be overridden in the Debug menu, right?

Reply:
This ability has just been added through #19645 👍

Reply:
🫡 thanks!